activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/process/BpmnModelServiceTaskCatchBoundaryValidator.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.modeling.validation.process;
+
+import static java.lang.String.format;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.activiti.bpmn.model.BoundaryEvent;
+import org.activiti.bpmn.model.BpmnModel;
+import org.activiti.bpmn.model.ErrorEventDefinition;
+import org.activiti.bpmn.model.ServiceTask;
+import org.activiti.cloud.modeling.api.ModelValidationError;
+import org.activiti.cloud.modeling.api.ValidationContext;
+
+/**
+ * Implementation of {@link BpmnCommonModelValidator} for validating service task boundaries
+ */
+public class BpmnModelServiceTaskCatchBoundaryValidator implements BpmnCommonModelValidator {
+
+    public static final String MISSING_BOUNDARY_WARNING = "Missing Catch Error boundary event";
+    public static final String INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION = "The service implementation on service '%s' might fail silently";
+    public static final String SERVICE_TASK_VALIDATOR_NAME = "BPMN service task catch boundary validator";
+
+    private final FlowElementsExtractor flowElementsExtractor;
+
+    private final ServiceTaskImplementationType[] serviceTaskImplementationTypes;
+
+    public BpmnModelServiceTaskCatchBoundaryValidator(FlowElementsExtractor flowElementsExtractor,
+        ServiceTaskImplementationType[] serviceTaskImplementationTypes) {
+        this.flowElementsExtractor = flowElementsExtractor;
+        this.serviceTaskImplementationTypes = Optional.ofNullable(serviceTaskImplementationTypes)
+            .orElse(new ServiceTaskImplementationType[0]);
+    }
+
+    @Override
+    public Stream<ModelValidationError> validate(BpmnModel bpmnModel,
+        ValidationContext validationContext) {
+
+        return flowElementsExtractor.extractFlowElements(bpmnModel, ServiceTask.class).stream()
+            .filter(serviceTask -> serviceTask.getImplementation() != null)
+            .map(serviceTask -> validateServiceTaskBoundary(serviceTask))
+            .filter(Optional::isPresent)
+            .map(Optional::get);
+    }
+
+    private Optional<ModelValidationError> validateServiceTaskBoundary(ServiceTask serviceTask) {
+        if (requiresBoundary(serviceTask)) {
+            return Optional.of(
+                new ModelValidationError(MISSING_BOUNDARY_WARNING,
+                    format(INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION,
+                        serviceTask.getId()), SERVICE_TASK_VALIDATOR_NAME, true));
+        }
+
+        return Optional.<ModelValidationError>empty();
+    }
+
+    private boolean requiresBoundary(ServiceTask serviceTask) {
```

Comment:
Maybe the method could be renamed to `requiredBoundaryIsMissing`, what do you think?

Reply:
It would be more clear and auto-descriptive. Done. #947